Add Solution for 002 add two numbers

yiru941013 opened the pull request:
@kuangami

yiru941013 committed:
Add Solution for 002 add two numbers

kuangami commented on java/002_Add_Two_Numbers.java:
```diff
@@ -0,0 +1,86 @@
+        prev = node;
+        
+        return head.next;
+    }
```
In the problem you need to save the head for output, so a dummy head is needed. But here for creating the list, you just need to output the head node. You can simply use the constructor `ListNode(int x) { val = x; }` to create a node.

kuangami commented on java/002_Add_Two_Numbers.java:
```diff
@@ -0,0 +1,86 @@
+        while (head.next != null)
+            head = head.next;
+        head.next = node;
+    }
```
The purpose of this ListNode class is to test your code. What we want here is to give your code two initial numbers. It's not necessary to add the element to the end of the list. If you add the element to the head of list, you can manipulate one element at a time, not going through the whole list. 

Here is what you do to initialize: 
2
2 -> 4
2 -> 4 -> 3

What I said is like this:
3
4 -> 3
2 -> 4 -> 3

You can use another constructor overriding the current one:
`ListNode(int x, ListNode node) { val = x; next = node; }`

yiru941013 committed:
Update 002_Add_Two_Numbers.java

delete the createList function

yiru941013 committed:
Update 002_Add_Two_Numbers.java

Use another constructor instead of a insertList function. 
Change the test code.

kuangami commented:
Excellent!
